### PR TITLE
Add PodMonitor Support with Metric Relabeling

### PR DESCRIPTION
**Pull request by wonderg**
### Description:
This PR introduces support for PodMonitor, enhancing the integration with Prometheus Operator. A key feature added is the ability to define metric relabeling configurations, giving users more flexibility in shaping the scraped metrics.

### Changes include:

- Updated README.md to reflect the new configurable parameter podmonitor.metricRelabelings.
- Added a new template for PodMonitor with provisions for metric relabeling.

### CODE REVIEW

**Comment by wonderg** [comment]
I've updated  charts/node-local-dns/templates/serviceaccount.yaml as well to fix possible logical issues
```
[ERROR] templates/serviceaccount.yaml: unable to parse YAML: invalid Yaml document separator: apiVersion: v1
```
The issue is with the --- separator at the beginning of the template combined with the conditional block {{- if .Values.serviceAccount.create -}}. If .Values.serviceAccount.create is set to false, Helm will render the template as an empty YAML file with just the --- separator, which is invalid.

To fix the issue, moved the separator --- inside the conditional block

File: charts/node-local-dns/templates/podmonitor.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+---
+apiVersion: monitoring.coreos.com/v1
+kind: PodMonitor
+metadata:
+  name: {{ include "node-local-dns.fullname" . }}
+  labels:
+    {{- include "node-local-dns.labels" . | nindent 4 }}
+    {{ toYaml .Values.podmonitor.promOperatorSelector | nindent 4 }}
+  annotations:
+  {{- with .Values.podmonitor.annotations }}
```
Comment (mjtrangoni) [inline charts/node-local-dns/templates/podmonitor.yaml:10]: @wonderg **with** action should be before `annotations:`

Comment (wonderg) [reply charts/node-local-dns/templates/podmonitor.yaml:9]: Agree, just fixed it. Please check

**Review by MonolithProjects — CHANGES_REQUESTED** [review]

**Comment by mjtrangoni** [comment]
@wonderg any news on this? I would really love to get all 3 open PR merged

**Comment by wonderg** [comment]
> 1

sure, please check recent commits.

**Review by MonolithProjects — CHANGES_REQUESTED** [review]
Hi @wonderg , it looks good. Just please bump the chart version to `1.6.0` in `charts/node-local-dns/Chart.yaml`

**Comment by wonderg** [comment]
> 1.6.0



> Hi @wonderg , it looks good. Just please bump the chart version to `1.6.0` in `charts/node-local-dns/Chart.yaml`

sure, just bumped it.

**Review by MonolithProjects — APPROVED** [review]
LGTM. Thanks @wonderg

**Pull request merged**